explainer.md

```diff
@@ -72,6 +72,113 @@ There are still many open questions, but a few big ones:
+>      personally-identifiable information (PII), or information derived from
+>      either?
+
+Yes, the API is designed to help facilitate communication of PII from secure wallet applications to verifier web servers. 
+
+> 04.  How do the features in your specification deal with sensitive information?
+
+Today online identity verification (eg. for KYC) is usually done by submitting photos of government identity documents (for example, uploading photos containing the PII via the web’s `<input type=file>` mechanism. Due to the privacy and security limits of this approach, credential issuers and verifiers are working to move to the use of wallet applications which can do selective disclosure of cryptographically attested properties. Today those approaches rely on generic web->app communication paths like custom schemes and server-to-server communication. 
+
+The use of wallets aims to improve on the status quo of uploading images in a number of ways:
+ * Enable the use of selective disclosure (eg. for more privacy-friendly age 18+ verification)
+ * Enable the PII to be end-to-end encrypted between the wallet application and the relevant verification server
+ * Enable the use of request authentication where a wallet is required to cryptographically verify that the requester has permission to access the credential
+
+Further, this specification aims to improve over the existing communication channels used by websites to talk to wallet apps by:
+ * Enabling browsers and operating systems to provide meaningful credential selection and permission screens to users prior to wallet applications becoming aware of the presentation request
+ * Securely communicate the origin of the requesting site to the wallet application so that it can implement it’s own MITM protection with support of WebPKI
+ * Enabling browsers to inspect requests and provide additional UI affordances to users
+
+> 05.  Do the features in your specification introduce state
+>      that persists across browsing sessions?
+
+No. In particular, the current scope is just about credential presentation (read-only). We have had requests to expand scope to consider credential issuance APIs but no work has begun on that.
+
+> 06.  Do the features in your specification expose information about the
+>      underlying platform to origins?
+
+Not directly. Wallets may (intentionally or inadvertently) expose some information through this API (such as some indication of which wallet app the user is using). But the primary motivation for these wallets over traditional federated authentication systems is that the wallet can act in the user’s interest to protect their privacy even from the credential issuer. Issuers generally choose which wallets to support their credentials in and can impose privacy requirements on those wallets. Users will often have their choice of multiple wallets and are expected to use reputation for privacy as part of their decision making.
```

Comment:
I would strike the sentence in here about what issuers will generally do, since I don't think it's true across all credential types (and perhaps wouldn't really even have a meaningful effect, if credentials are thereafter shared with any RP of choice, e.g., "shared with another wallet acting as an RP").

Issuers can and do certainly make wallet suggestions, but focusing on the user making the ultimate choices here seems better.

```suggestion
Not directly. Wallets may (intentionally or inadvertently) expose some information through this API (such as some indication of which wallet app the user is using). But the primary motivation for these wallets over traditional federated authentication systems is that the wallet can act in the user’s interest to protect their privacy even from the credential issuer. Users will often have their choice of multiple wallets and are expected to use reputation for privacy as part of their decision making.
```

Reply:
As a user, I do not want to have 25 credential issuers where each may require a different wallet. Physical wallets are chosen strictly by their users, and they contain whatever physical credentials the user wants to insert. Digital wallets should have roughly the same characteristics.

Reply:
@TallTed,

Agreed. Nevermind the potential tracking problem with having each issuer require use of their own app. Such an approach would really just be the two party model again and rely on trust in software instead of credentials and cryptographic proofs, recreating problems that the three party model was designed to mitigate.

Reply:
As much as that might be what we'd like to happen, that doesn't seem to be what's happening in some major deployments such as EUDI right? Under eIDAS each memeber state will, I believe, certify one or more wallets and certainly not support issuing EUDI credentials into any wallet of the user's choosing.

Nevertheless I agree that users should come first, so I've swapped them. I've also relaxed "generally" to "may".